src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -553,7 +553,7 @@ private void CheckDisposed()
                     }
                 }
             }
-            catch (Exception ex)
+            catch (Exception ex) when (!(ex is HubException))
```

Comment:
When would this ever happen?

Reply:
When you received a handshake response with an `"error"` set

Reply:
That seems like a product bug no?

Reply:
Ah I see, this code looks like it needs to be reworked, it's throwing and logging then rethrowing...

Reply:
Add a comment please